resolwe_runtime_utils.py

```diff
@@ -275,14 +301,224 @@ def checkrc(rc, *args):
+            with open(src, 'rb') as f_in, gzip.open(file_name + '.gz', 'wb') as f_out:
+                shutil.copyfileobj(f_in, f_out, CHUNK_SIZE)
+
+        if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+            try:
+                shutil.copyfile(src, file_name)
+            except shutil.SameFileError:
+                pass  # Skip copy of downloaded files
+
+        return (
+            file_name + '.gz'
+            if imported_format == ImportedFormat.COMPRESSED
+            else file_name
+        )
+
+    # Large file download from Google Drive requires cookie and token.
+    response = None
+    if re.match(
+        r'^https://drive.google.com/[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]$',
+        src,
+    ):
+        session = requests.Session()
+        response = session.get(src, stream=True)
+
+        token = None
+        for key, value in response.cookies.items():
+            if key.startswith('download_warning'):
+                token = value
+                break
+
+        if token is not None:
+            params = {'confirm': token}
+            response = session.get(src, params=params, stream=True)
+
+    elif re.match(
+        r'^(https?|ftp)://[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]$', src
+    ):
+        response = requests.get(src, stream=True)
+
+    if response:
+        with open(file_name, 'wb') as f:
+            total = response.headers.get('content-length')
+            total = float(total) if total else None
+            downloaded = 0
+            current_progress = 0
+            for content in response.iter_content(chunk_size=CHUNK_SIZE):
+                f.write(content)
+
+                if total is not None and progress_to is not None:
+                    downloaded += len(content)
+                    progress_span = progress_to - progress_from
+                    next_progress = progress_from + progress_span * total / downloaded
+                    next_progress = round(next_progress, 2)
+
+                    if next_progress > current_progress:
+                        print(progress(current_progress))
+                        current_progress = next_progress
+
+        # Check if a temporary file exists.
+        if not os.path.isfile(file_name):
+            print(
+                error("Download failed: temporary file not found {}".format(file_name))
+            )
+
+        src = file_name
+    else:
+        if not os.path.isfile(src):
+            print(error("Source file not found {}".format(src)))
+
+    # Decide which import should be used.
+    if re.search(r'\.(bz2|zip|rar|7z|tgz|tar\.gz|tar\.bz2)$', file_name):
+        destination_file_name = import7z()
+    elif file_name.endswith('.gz'):
+        destination_file_name = importGz()
+    else:
+        destination_file_name = importUncompressed()
+
+    if progress_to is not None:
+        print(progress(progress_to))
+
+    return destination_file_name
```

Comment:
The case of overwriting an existing file named `destination_file_name` is unhandled. This can occur in processes with multiple `basic:file` inputs where the user selects local files with the same file name in different directories. In processes we normally keep the same file name as on user's disk.

We haven't noticed such cases yet though.

Reply:
True. This is problematic. Python process will crash if this happens, but most likely with an ugly error.

Since there are no known occurrences of this issue, we may want to fix it in the future.